mobile/mobile_test.go

```diff
@@ -211,23 +208,26 @@ func TestMobile_PrepareFiles(t *testing.T) {
 		t.Errorf("prepare files failed: %s", err)
 		return
 	}
-	dir := core.Directory{}
-	if err := json.Unmarshal([]byte(res), &dir); err != nil {
+	pre := new(pb.MobilePreparedFiles)
+	if err := proto.Unmarshal(res, pre); err != nil {
 		t.Error(err)
 		return
 	}
-	if len(dir) != 6 {
+	if len(pre.Dir.Files) != 6 {
 		t.Error("wrong number of files")
 	}
-	prepared = res
+	dir, err = proto.Marshal(pre.Dir)
```

Comment:
`AddThreadFile` takes marshalled `Directory` bytes, which is one field from the prepared result.